contracts/external/policies/ERC20SpendingLimitPolicy.sol

```diff
@@ -128,23 +101,139 @@ contract ERC20SpendingLimitPolicy is IActionPolicy {
         returns (uint256)
     {
         if (value != 0) return VALIDATION_FAILED;
-        (bool isTokenTransfer, uint256 amount) = _isTokenTransfer(account, callData);
+        (bool isTokenTransfer, uint256 amount) = _isTokenTransferOrApprove(account, callData);
         if (!isTokenTransfer) return VALIDATION_FAILED;
 
         TokenPolicyData storage $ = _getPolicy({ id: id, userOpSender: account, token: target });
 
-        uint256 spendingLimit = $.spendingLimit;
-        uint256 alreadySpent = $.alreadySpent;
+        if (
+            bytes4(callData[0:4]) == IERC20.approve.selector
+                || bytes4(callData[0:4]) == bytes4(keccak256("increaseAllowance(address,uint256)"))
+        ) {
+            // increase approval case
+            // if the amount is uint256 max (max alowance) and the $.approvedAmount is not 0,
+            // it will overflow and revert. In theory this is not what should happen as the approval in erc20
+            // contract can be successfully set to max in this case, and if the previous approval was not yet spent,
+            // can be a valid case, however accounting for it will cause so many other edge cases and security
+            // considerations
+            // that we decided to stick with this approach and document the fact that the session key should never
+            // operate with max allowances
+            // in fact it has no reasons of doing this as the purpose of session key is that it can issue whatever
+            // ammount of signature, without
+            // making UX worse for user. So it is recommended that a session key always permits the exact amount of
+            // tokens that is about to be spent by spender.
+            $.approvedAmount += amount;
```

Comment:
if the selector is approve not increaseAllowance then this would overreport the approvals in some cases right (eg: current approval is 10, new is 20 but total here is 30 rather than 20)

think we discussed this before but just wanted to point it out again

Reply:
yep, this is what we decided to go with, as being extra secure vs less secure.
Because we do not know if the allowance was spent or not, if this allowance of 10 was spent in the erc20 contract, then adding 20 to 10 is correct to do. The only real issue here is adding max allowance to non-0 one, but this is not recommended flow (max allowance for a session key) anyways

Reply:
yeah ig I was wondering if we should differentiate using `approve` and using `increaseAllowance`

Reply:
I think we discussed this and decided not to? :)